src/multivariate/composite.jl

```diff
@@ -0,0 +1,203 @@
+# Composite Multivariate distribution
+
+## Generic Composite multivariate distribution class
+
+module CompositeDistributions
+
+if !isdefined(:Distributions) using Distributions end
+
+import Base.length, Base.mean, Base.show
+import Distributions.params
+import Distributions.mean, Distributions.mode, Distributions.var, Distributions.cov
+import Distributions.entropy, Distributions.insupport
+import Distributions._logpdf, Distributions._pdf!
+import Distributions._rand!
+
+export AbstractCompositeContinuousDist, ContinuousMultivariateDistribution
+export GenericCompositeContinuousDist, CompositeDist
+export length, params, set_params!, mean, mode, var, cov, entropy
+export insupport, _logpdf, gradlogpdf, _rand!
+
+
+abstract AbstractCompositeContinuousDist <: ContinuousMultivariateDistribution
+#abstract AbstractCompositeDiscreteDist <: DiscreteMultivariateDistribution  # An idea, but not implemented yet
+
+immutable GenericCompositeContinuousDist <: AbstractCompositeContinuousDist
+    dist::Vector{ContinuousDistribution}
```

Comment:
I think this should be a tuple. Were it to remain a vector, you'd still want this type to be a parametric type since the current representation of this type creates a quasi-black-box that type inference can't fully see into. Usually this sort of thing looks like:

```julia
immutable GenericCompositeContinuousDist{T <: ContinuousDistribution} <: AbstractCompositeContinuousDist
    dist::Vector{T}
[...]
```